odoo/models.py

```diff
@@ -3025,25 +3025,31 @@ def _update_field_translations(self, field_name, translations, digest=None):
             # a non-related non-stored computed field cannot be translated, even if it has inverse function
             return False
 
+        # Strictly speaking, a translated related/computed field cannot be stored
+        # because the compute function only support one language
+        # `not field.store` is a redundant logic.
+        # But some developers store translated related fields.
+        # In these cases, only all translations of the first stored translation field will be updated
+        # For other stored related translated field, the translation for the flush language will be updated
+        if field.related and not field.store:
+            related_path, field_name = field.related.rsplit(".", 1)
+            return self.mapped(related_path)._update_field_translations(field_name, translations, digest)
+        self.check_access_rights('write')
+        self.check_field_access_rights('write', [field_name])
+        self.check_access_rule('write')
+
         if field.translate is True:
-            for lang, translation in translations.items():
-                if translation is not None:
-                    self.with_context(lang=lang)[field_name] = translation
+            # empty string translation and Falsy translation are used to void the corresponding translation
+            translations = {
+                lang: None if not translation and lang != 'en_US' else str(translation)
+                for lang, translation in translations.items()
+            }
+            self.invalidate_recordset([field_name])
+            self._cr.execute(f'''
+                UPDATE {self._table} SET {field_name} = jsonb_strip_nulls({field_name} || %s)
```

Comment:
Ahem, this updates **all** records in the table!  I think you forgot the WHERE clause :wink: